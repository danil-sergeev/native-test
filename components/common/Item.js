import React from "react";
import {Text, TouchableWithoutFeedback} from "react-native";

export default function Item({ id, text, pressed, onPress, style }) {
    const backgroundColor = pressed ? '#FD7F4A' : '#ffffff';
    return (
        <TouchableWithoutFeedback
            onPress={() => onPress(id)}
        >
            <Text
                style={[style, { backgroundColor }]}
            >
                {text}
            </Text>
        </TouchableWithoutFeedback>
    )
}