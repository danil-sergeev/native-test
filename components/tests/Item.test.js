import React from "react";
import renderer from "react-test-renderer";

import Item from "../common/Item";

it(`renders correctly`, () => {
    const tree = renderer.create(
        <Item id='1' text='test' pressed='false' onPress={() => {}}>Snapshot test!</Item>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it(`changes color when pressed`, () => {
    const tree = renderer.create(
        <Item id='1' text='test' pressed='true' onPress={() => {}}>Snapshot test!</Item>
    ).toJSON();
    const [_, {backgroundColor}] = tree.props.style;
    expect(backgroundColor === "#FD7F4A").toEqual(true);
});