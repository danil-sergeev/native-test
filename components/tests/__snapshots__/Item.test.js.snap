// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FD7F4A",
      },
    ]
  }
>
  test
</Text>
`;
